dll/win32/shell32/CUserNotification.h

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2018 Hermes Belusca-Maito
+ *
+ * Pass on icon notification messages to the systray implementation
+ * in the currently running shell.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#ifndef _USERNOTIFICATION_H_
+#define _USERNOTIFICATION_H_
+
+#undef PlaySound
+
+class CUserNotification :
+    public CComCoClass<CUserNotification, &CLSID_UserNotification>,
+    public CComObjectRootEx<CComMultiThreadModelNoCS>,
+    public IUserNotification
```

Comment:
Why not use `CWindowImpl`? You can use the `m_thunk` then for the WorkerWndProc, without having to toy around with `GetWindowLongPtr` etc

Reply:
I don't need the full-fledged functionality that CWindowImpl offers, I just need a single worker window (from "WorkerW" class to be precise if I want to be compatible wrt. Windows) to run my notification window proc.